src/android/jni/jni_alohalytics.cc

```diff
@@ -96,20 +117,42 @@ JNIEXPORT void JNICALL Java_org_alohalytics_Statistics_logEvent__Ljava_lang_Stri
 
 JNIEXPORT void JNICALL Java_org_alohalytics_Statistics_logEvent__Ljava_lang_String_2_3Ljava_lang_String_2(
     JNIEnv* env, jclass, jstring eventName, jobjectArray keyValuePairs) {
-  const jsize count = env->GetArrayLength(keyValuePairs);
-  TStringMap map;
-  string key;
-  for (jsize i = 0; i < count; ++i) {
-    const jstring jni_string = static_cast<jstring>(env->GetObjectArrayElement(keyValuePairs, i));
-    if ((i + 1) % 2) {
-      key = ToStdString(env, jni_string);
-      map[key] = "";
-    } else {
-      map[key] = ToStdString(env, jni_string);
-    }
-    if (jni_string) env->DeleteLocalRef(jni_string);
+  LogEvent(ToStdString(env, eventName), FillMapHelper(env, keyValuePairs));
+}
+
+JNIEXPORT void JNICALL
+    Java_org_alohalytics_Statistics_logEvent__Ljava_lang_String_2_3Ljava_lang_String_2ZJDDFZDZFZFB(
+        JNIEnv* env,
+        jclass,
+        jstring eventName,
+        jobjectArray keyValuePairs,
+        jboolean hasLatLon,
+        jlong timestamp,
+        jdouble lat,
+        jdouble lon,
+        jfloat accuracy,
+        jboolean hasAltitude,
+        jdouble altitude,
+        jboolean hasBearing,
+        jfloat bearing,
+        jboolean hasSpeed,
+        jfloat speed,
+        jbyte source) {
+  alohalytics::Location l;
+  if (hasLatLon) {
```

Comment:
14900000 vs 17000000 results in Google, I never heard about latlong before, and it is 4 vs 3 letters (slightly inconsistent)